includes/woopay/class-woopay-session.php

```diff
@@ -542,6 +541,11 @@ private static function has_valid_request_signature() {
 	 * @return bool True if WooPay is enabled, false otherwise.
 	 */
 	private static function is_woopay_enabled(): bool {
+		// There were previously instances of this function being called too early. While those should be resolved, adding this defensive check as well.
+		if ( ! class_exists( WC_Payments_Features::class ) || ! class_exists( WC_Payments::class ) || is_null( WC_Payments::get_gateway() ) ) {
+			return false;
+		}
```

Comment:
Love this addition 👏 ❤️

Reply:
Thanks, but credit here goes to @ricardo for his proposed fix in #7005. 👏 I loved the defensive programming here as well!